db/commit.go

```diff
@@ -49,6 +51,7 @@ type Local struct {
 	Date       datetime.DateTime
 	Name       string
 	Args       types.Value
+	Original   hash.Hash
```

Comment:
[fix] The plan we talked about was to make this field `types.Ref`. I still like that plan because it makes the reference "strong" in Noms terms (so sync will be aware of it). The way to make it optional would be as so:

```
type Local struct {
  ...
  Original types.Ref `noms:",omitempty"
}
```

Reply:
the plan changed when i forgot that it was important to keep a strong reference and was under the mistaken impression that you can't have zero refs (in which case we would have to keep a separate replay commit type). i had added a ref to the local commit and when it wasn't set en/decoding would panic. i somehow forgot about omitempty, doh. will fix it, thanks.

Reply:
It's ok, it's a lot. It's all super fiddly. This is part of why I like the idea of eventually pulling out a much smaller subset of Noms into a separate project.

Anyway, you can have zero instances of things in `noms/types`, but they don't work very well. Pretty much all you can do with them is ask them if they are `.IsZeroValue()`. If you try to do other things with them they will fail cryptically. Importantly you can't save them in Noms, which is why you have to pair them with `noms:",omitempty"` so the marshaling code will skip over them.